examples/getAccountInfo.cpp

```diff
@@ -2,112 +2,100 @@
 #include <spdlog/spdlog.h>
 
 #include <nlohmann/json.hpp>
+
+#include "solana.hpp"
 using json = nlohmann::json;
 
 #include "mango_v3.hpp"
 
+const std::string DEFAULT_PUBLICKEY = "11111111111111111111111111111111";
+
 int main() {
   std::string rpc_url =
       "https://mango.rpcpool.com/946ef7337da3f5b8d3e4a34e7f88";
   auto connection = solana::rpc::Connection(rpc_url);
-  const json req = connection.getAccountInfoRequest(
-      "98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue");
-  const std::string jsonSerialized = req.dump();
-  spdlog::info("REQ: {}", jsonSerialized);
-
-  cpr::Response r =
-      cpr::Post(cpr::Url{rpc_url}, cpr::Body{jsonSerialized.c_str()},
-                cpr::Header{{"Content-Type", "application/json"}});
-
-  if (r.status_code == 0 || r.status_code >= 400) {
-    spdlog::error("Error: {}, {}", r.status_code, r.error.message);
-    return 1;
-  } else {
-    spdlog::info("RES: {}", r.text);
-    json res = json::parse(r.text);
-
-    const std::string encoded = res["result"]["value"]["data"][0];
-    const std::string decoded = solana::b64decode(encoded);
-    const mango_v3::MangoGroup *group =
-        reinterpret_cast<const mango_v3::MangoGroup *>(decoded.data());
-    spdlog::info("DEC:");
-    spdlog::info("numOracles: {}", group->numOracles);
-
-    for (int i = 0; i < mango_v3::MAX_TOKENS; ++i) {
-      if (i >= group->numOracles && i != mango_v3::QUOTE_INDEX) continue;
-
-      const auto &token = group->tokens[i];
-      const auto mintPk = token.mint.toBase58();
-
-      if (mintPk == std::string("11111111111111111111111111111111")) continue;
-
-      const auto rootBankPk = token.rootBank.toBase58();
-      if (i != mango_v3::QUOTE_INDEX) {
-        spdlog::info("TOK: {}", i);
-      } else {
-        spdlog::info("QUOTE: {}", i);
-      }
-      spdlog::info("  mint: {}", mintPk);
-      spdlog::info("  rootBank: {}", rootBankPk);
-      spdlog::info("  decimals: {}", static_cast<int>(token.decimals));
+  const std::string account = "98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue";
+  const mango_v3::MangoGroup grp =
+      connection
+          .getAccountInfo<mango_v3::MangoGroup>(
+              solana::PublicKey::fromBase58(account))
+          .value.data;
+  const auto group = &grp;
+
+  spdlog::info("DEC:");
+  spdlog::info("numOracles: {}", group->numOracles);
+
+  for (int i = 0; i < mango_v3::MAX_TOKENS; ++i) {
+    if (i >= group->numOracles && i != mango_v3::QUOTE_INDEX) continue;
+
+    const auto &token = group->tokens[i];
+    const auto mintPk = token.mint.toBase58();
+
+    if (mintPk == std::string(DEFAULT_PUBLICKEY)) continue;
```

Comment:
why do you call a copy constructor?

Reply:
don't need this to be perfect, just an example. so let's move on, no need for further changes:
just use PublicKey::default and PublicKey::operator== for future checks.